### PR TITLE
Modernize build

### PR DESCRIPTION
**Pull request by basil**
Modernizes this plugin's build/test toolchain to conform to our latest best practices from:

- https://github.com/jenkinsci/archetypes
- https://github.com/jenkinsci/incrementals-tools

### CODE REVIEW

File: .mvn/extensions.xml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<extensions xmlns="http://maven.apache.org/EXTENSIONS/1.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/EXTENSIONS/1.0.0 http://maven.apache.org/xsd/core-extensions-1.0.0.xsd">
+  <extension>
+    <groupId>io.jenkins.tools.incrementals</groupId>
+    <artifactId>git-changelist-maven-extension</artifactId>
+    <version>1.7</version>
+  </extension>
+</extensions>
```
Comment (basil) [inline .mvn/extensions.xml:7]: See https://github.com/jenkinsci/incrementals-tools

File: .mvn/maven.config [hunk 2]
```diff
@@ -0,0 +1,2 @@
+-Pconsume-incrementals
+-Pmight-produce-incrementals
```
Comment (basil) [inline .mvn/maven.config:2]: See https://github.com/jenkinsci/incrementals-tools

File: Jenkinsfile [hunk 3]
```diff
@@ -1,6 +1,6 @@
 buildPlugin(
   useContainerAgent: true,
   configurations: [
-    [platform: 'linux', jdk: 11],
-    [platform: 'windows', jdk: 11],
+    [platform: 'linux', jdk: 21],
+    [platform: 'windows', jdk: 17],
```
Comment (basil) [inline Jenkinsfile:5]: As per latest recommendations in the archetype: https://github.com/jenkinsci/archetypes

File: pom.xml [hunk 4]
```diff
@@ -19,27 +20,27 @@
     <parent>
         <groupId>org.jenkins-ci.plugins</groupId>
         <artifactId>plugin</artifactId>
-        <version>4.66</version>
+        <version>4.75</version>
```
Comment (basil) [inline pom.xml:23]: Latest release at the time of this writing.

File: pom.xml [hunk 5]
```diff
@@ -19,27 +20,27 @@
...
         <relativePath />
     </parent>
 
     <artifactId>fortify</artifactId>
-    <version>23.1.40-SNAPSHOT</version>
+    <version>${revision}${changelist}</version>
```
Comment (basil) [inline pom.xml:28]: See https://github.com/jenkinsci/incrementals-tools

File: pom.xml [hunk 6]
```diff
@@ -19,27 +20,27 @@
...
     <packaging>hpi</packaging>
 
     <properties>
-        <gitHubRepo>jenkinsci/fortify-plugin</gitHubRepo>
-        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <revision>23.1.40</revision>
+        <changelist>-SNAPSHOT</changelist>
```
Comment (basil) [inline pom.xml:33]: See https://github.com/jenkinsci/incrementals-tools

File: pom.xml [hunk 7]
```diff
@@ -19,27 +20,27 @@
...
+        <gitHubRepo>jenkinsci/${project.artifactId}-plugin</gitHubRepo>
```
Comment (basil) [inline pom.xml:34]: As per latest recommendations in the archetype: https://github.com/jenkinsci/archetypes

File: pom.xml [hunk 8]
```diff
@@ -19,27 +20,27 @@
...
     </parent>
 
     <artifactId>fortify</artifactId>
-    <version>23.1.40-SNAPSHOT</version>
+    <version>${revision}${changelist}</version>
     <packaging>hpi</packaging>
 
     <properties>
-        <gitHubRepo>jenkinsci/fortify-plugin</gitHubRepo>
-        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
```
Comment (basil) [inline pom.xml:32]: The latest plugin parent POM already sets this, so no need to duplicate it here.

File: pom.xml [hunk 9]
```diff
@@ -19,27 +20,27 @@
...
+        <revision>23.1.40</revision>
+        <changelist>-SNAPSHOT</changelist>
+        <gitHubRepo>jenkinsci/${project.artifactId}-plugin</gitHubRepo>
         <jenkins.version>2.387.3</jenkins.version>
         <bom>2.387.x</bom>
-        <bom.version>2423.vce598171d115 </bom.version>
-        <disabledTestInjection>true</disabledTestInjection>
```
Comment (basil) [inline pom.xml:36]: No need to disable `InjectedTest`, as it is currently passing.

File: pom.xml [hunk 10]
```diff
@@ -19,27 +20,27 @@
...
+        <bom.version>2543.vfb_1a_5fb_9496d</bom.version>
```
Comment (basil) [inline pom.xml:37]: Latest BOM for the 2.387.x line at the time of this writing.

File: pom.xml [hunk 11]
```diff
@@ -19,27 +20,27 @@
...
         <hpi.compatibleSinceVersion>20.2.34</hpi.compatibleSinceVersion>
     </properties>
 
     <name>Fortify</name>
     <description>Adds the ability to perform security analysis with Fortify Static Code Analyzer, upload results to Software Security Center, show analysis results summary, and set build failure criteria based on analysis results.</description>
-    <url>https://github.com/jenkinsci/fortify-plugin</url>
+    <url>https://github.com/jenkinsci/${project.artifactId}-plugin</url>
```
Comment (basil) [inline pom.xml:43]: As per latest recommendations in the archetype: https://github.com/jenkinsci/archetypes

File: pom.xml [hunk 12]
```diff
@@ -50,9 +51,9 @@
 
     <scm>
         <connection>scm:git:https://github.com/${gitHubRepo}.git</connection>
-        <developerConnection>scm:git:https://github.com/${gitHubRepo}.git</developerConnection>
+        <developerConnection>scm:git:git@github.com:${gitHubRepo}.git</developerConnection>
         <url>https://github.com/${gitHubRepo}</url>
-        <tag>v22.2.39</tag>
+        <tag>${scmTag}</tag>
```
Comment (basil) [inline pom.xml:56]: As per latest recommendations in the archetype: https://github.com/jenkinsci/archetypes

File: pom.xml [hunk 13]
```diff
@@ -73,11 +74,6 @@
             <id>repo.jenkins-ci.org</id>
             <url>https://repo.jenkins-ci.org/public/</url>
         </repository>
-        <repository>
-            <id>maven.apache.org</id>
-            <name>fortify</name>
-            <url>https://repo.maven.apache.org/maven2</url>
-        </repository>
```
Comment (basil) [inline pom.xml:80]: This is already set in the Maven default POM, so no need to set it again here.

File: pom.xml [hunk 14]
```diff
@@ -173,20 +169,30 @@
         <dependency>
             <groupId>io.jenkins.plugins</groupId>
             <artifactId>jakarta-mail-api</artifactId>
-            <version>2.0.1-3</version>
```
Comment (basil) [inline pom.xml:176]: Version already managed in the plugin BOM, so no need to set it here.

File: pom.xml [hunk 15]
```diff
@@ -173,20 +169,30 @@
...
+                            <rules>
+                                <requireUpperBoundDeps>
+                                    <excludes combine.children="append">
+                                        <exclude>com.squareup.okio:okio-jvm</exclude>
+                                        <exclude>org.jetbrains.kotlin:kotlin-stdlib-common</exclude>
+                                        <exclude>org.jetbrains.kotlin:kotlin-stdlib-jdk8</exclude>
+                                    </excludes>
+                                </requireUpperBoundDeps>
+                            </rules>
+                        </configuration>
```
Comment (basil) [inline pom.xml:195]: Disabling Enforcer completely suppress all errors, even new violations. This change re-enables Enforcer but suppresses only existing violations, meaning that new violations will be enforced going forward. The long-term solution, described in the preceding comment, is to switch to a library plugin instead.

**Comment by akaryakina** [comment]
Thanks @basil. We greatly appreciate your time and help!

**Review by akaryakina — COMMENTED** [review]
LGTM

**Pull request merged**